Improve status reporting

alaypatel07 opened the pull request:
Addresses #7

alaypatel07 committed:
Add exception handling to lookup plugin

alaypatel07 committed:
Update reconcile role to improve status reporting

johnkim76 commented on roles/reconcile/tasks/reconcile_pods.yaml:
```diff
@@ -23,27 +48,40 @@
   set_fact:
     etcd_cluster_members: "{{ members }}"
   when: (etcd_cluster_phase == "existing" and etcd_secure_client == "present")
+  ignore_errors: yes
```
just curious, were these `ignore_errors` the only solution? or may be the best way to do it?

alaypatel07 replied:
Yeah couldn't any other way!

alaypatel07 committed:
Update status when scaling, add readiness probe

alaypatel07 committed:
Update the dns names of pods to avoid dns resolution problems.

With just the {HOSTNAME}.${SET_NAME}, the kube-dns was resolving
different ip addresses leading to problems like:
```
2019-03-22 18:31:23.676989 I | pkg/netutil: resolving example-etcd-cluster-5.example-etcd-cluster:2380 to 198.105.244.228:2380
2019-03-22 18:31:23.677718 I | pkg/netutil: resolving example-etcd-cluster-5.example-etcd-cluster:2380 to 172.17.0.12:2380
```

The init script now checks if this pod is scaling, and then check if this pods is joining back. This avoids problems of failures during scaling
